File: testfeature-noop/src/main/java/com/luigivampa92/testfeature_impl/TestFeatureHolder.java
package com.luigivampa92.testfeature_impl;

//import androidx.annotation.NonNull;

import com.luigivampa92.testfeature_api.TestFeatureApi;
import com.luigivampa92.testfeature_api.TestFeatureAvailabilityChecker;
import com.luigivampa92.testfeature_api.TestFeatureLauncher;

public class TestFeatureHolder implements TestFeatureApi {

//    @NonNull
    @Override
    public TestFeatureAvailabilityChecker getAvailabilityChecker() {
        return new TestFeatureAvailabilityChecker() {
            @Override
            public boolean isEnabled() {
                return false;
            }
        };
    }

//    @NonNull
    @Override
    public TestFeatureLauncher getFeatureLauncher() {
        return new TestFeatureLauncher() {
            @Override
            public void launch() {}
        };
    }
}

File: testfeature-impl/src/main/java/com/luigivampa92/testfeature_impl/TestFeatureHolder.kt
package com.luigivampa92.testfeature_impl

import com.luigivampa92.testfeature_api.TestFeatureApi
import com.luigivampa92.testfeature_api.TestFeatureAvailabilityChecker
import com.luigivampa92.testfeature_api.TestFeatureLauncher

class TestFeatureHolder : TestFeatureApi {

    override fun getAvailabilityChecker(): TestFeatureAvailabilityChecker {
        return TestFeatureAvailabilityCheckerImpl()
    }

    override fun getFeatureLauncher(): TestFeatureLauncher {
        return TestFeatureLauncherImpl()
    }
}